packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_prop/string_prop.rs

```diff
@@ -115,6 +185,20 @@ impl PropUpdater<String, RequiredData> for StringProp {
         _is_direct_change_from_action: bool,
     ) -> Result<Vec<DependencyValueUpdateRequest>, InvertError> {
         match data.strings.len() {
+            0 => {
+                if self.propagate_came_from_default {
+                    // if propagate_came_from_default is true,
+                    // then always_return_value is true on the string data query,
+                    // so we should never reach this
```

Comment:
I don't understand why, but I believe you...

Reply:
I tried making the wording more explicit.